### PR TITLE
[public-api] Expose WorkspaceInstance as part of Workspace message

### PR DESCRIPTION
**Pull request by easyCZ**
## Description

When making a `GetWorkspace` call, we don't currently have a way to provide the WorkspaceInstance (and information) to the caller. 

A `WorkspaceInstance` is the result of a `WorkspaceCreate/Start` and as such it is reflected in the `status` part of the Workspace. Similar to how child k8s resources are references in k8s status object.

## Related Issue(s)

Fixes #

## Release Notes

```release-note
```

## Documentation

/uncc
/hold

### CODE REVIEW

**Comment by roboquat** [comment]
@easyCZ: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by csweichel** [comment]
A few thoughts:
- it does make the API easier, and helps folks understand what's the relationship between workspace and instance. Today we say that a "workspace has failed" and really mean that the workspace instance has failed - making the instance part of the status captures this intuition.
- we have heaps of calls which do not require the instance, i.e. we'd be retrieving the instance in case where it's not needed today. That said, by virtue of caching and fieldmasks (down the road) this is less of a problem.
- I like wrapping the instance in a `WorkspaceStatus`, because we have other bits of workspace status once we make workspace creation async.

TLDR; let's do this

**Comment by easyCZ** [comment]
@laushinka This will allow us to expose more information as part of the GetWorkspace() call.

**Comment by easyCZ** [comment]
In this PR, we're only adding the protos and the implementation will follow in a sub-sequent PR. There is no usage and proto guarantees backwards compatibility in this case so this is safe to land. Just needs an approval.

**Comment by laushinka** [comment]
Happy to approve this, @easyCZ! Just needs to be rebased 🙏🏽
